### PR TITLE
Add `arro3`

### PR DESCRIPTION
**Pull request by kylebarron**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/arro3/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+arro3_module:
+  - arro3-core
+  # - arro3-compute
+  # - arro3-io
```
Comment (kylebarron) [inline recipes/arro3/conda_build_config.yaml:4]: Question for maintainers @conda-forge/help-python: arro3 is a namespace package, so there will be multiple internal modules that are published separately, but expected to interoperate and to all use the same version. There may be more namespace modules in the future. Is it ok to have a single conda-forge recipe for all modules? Or do I need to have a different recipe for each namespace module.

**Comment by kylebarron** [comment]
@conda-forge/help-python Can I get a review on this?

And any thoughts on this:

> arro3 is a namespace package, so there will be multiple internal modules that are published separately, but expected to interoperate and to all use the same version. There may be more namespace modules in the future. Is it ok to have a single conda-forge recipe for all modules? Or do I need to have a different recipe for each namespace module.

**Comment by kylebarron** [comment]
@conda-forge/help-python This is ready for review.

Can I get feedback on this question?

> arro3 is a namespace package, so there will be multiple internal modules that are published separately, but expected to interoperate and to all use the same version. There may be more namespace modules in the future. Is it ok to have a single conda-forge recipe for all modules? Or do I need to have a different recipe for each namespace module.

Since it takes so long to publish a new package recipe to conda-forge, I would prefer to have a single `arro3` recipe that publishes all namespace packages (i.e. publishes `arro3-core`, `arro3-io`, `arro3-compute`, and so on)

File: recipes/arro3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{% set version = "0.2.1" %}
+
+package:
+  name: {{ arro3_module }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/kylebarron/arro3/archive/refs/tags/py-v0.2.1.tar.gz
```
Comment (ocefpaf) [inline recipes/arro3/meta.yaml:8]: We need this to get the autoupdate bot.

```suggestion
  url: https://github.com/kylebarron/arro3/archive/refs/tags/py-v{{ version }}.tar.gz
```

**Comment by ocefpaf** [comment]
> Since it takes so long to publish a new package recipe to conda-forge, I would prefer to have a single `arro3` recipe that publishes all namespace packages (i.e. publishes `arro3-core`, `arro3-io`, `arro3-compute`, and so on)

They are all in the same source, right? I think it is OK. It is not too different from a multiple output recipe.

**Comment by kylebarron** [comment]
Thanks for the review!

> They are all in the same source, right? I think it is OK. It is not too different from a multiple output recipe.

Yes; they are all in the same source. Since the other namespace modules depend on arro3-core, I need to first get arro3-core published, and then once I get access to the feedstock, I can publish the other namespace moduels.

**Comment by kylebarron** [comment]
@conda-forge/help-python, ready for review!

**Comment by kylebarron** [comment]
Thank you!

**Comment by kylebarron** [comment]
@conda-forge/help-python, @ocefpaf it looks like there was an error in creating the feedstock. The https://github.com/conda-forge/arro3-core-feedstock repo was _created_ but there are no commits there.

I'm guessing there might be issues in trying to distribute namespace packages from a single recipe?

**Comment by ocefpaf** [comment]
I guess that the right way to do this is to have a single feedstock and use multiple outputs.

**Comment by kylebarron** [comment]
That was what I intended to do; but I first needed to publish `arro3-core` before I could publish the other outputs that depended on `arro3-core`. 

It seems I might need to go the other route; make a separate feedstock for each namespace package...

**Comment by ocefpaf** [comment]
> It seems I might need to go the other route; make a separate feedstock for each namespace package...

Remove the conda-build yaml file first. That should probably fix the feedstock creation. then you can work in the feedstock to add the rest.

**Pull request merged**